components/script/dom/bindings/str.rs

```diff
@@ -322,10 +322,79 @@ impl DOMString {
     }
 
     /// A valid month string should be "YYYY-MM"
-    /// YYYY must be four or more digits, MM both must be two digits
+    /// YYYY must be four or more digits, where YYYY > 0
+    /// MM both must be two digits, where 1 <= MM <= 12
     /// https://html.spec.whatwg.org/multipage/#valid-month-string
     pub fn is_valid_month_string(&self) -> bool {
-        parse_month_string(&self.0).is_ok()
```

Comment:
Whoops, I missed reading this commit. While in general I'm in favour of using state machines for clarity of purpose and avoiding invalid states, I find this one to be difficult to follow. I'm not sure if it's the names of the states, or the repetition between them, or something else, but my eyes keep glazing over when I try to follow the logic. Given how straightforward the specification text is, I'm not exactly sure what to recommend.

Reply:
My reading of the specification is that the parsing algorithm for month strings is the same as the validation algorithm for month strings, so I don't see a good reason to reimplement this particular algorithm. Presumably there are other input types where they diverge?

Reply:
> My reading of the specification is that the parsing algorithm for month strings is the same as the validation algorithm for month strings, so I don't see a good reason to reimplement this particular algorithm. Presumably there are other input types where they diverge?

Actually, the tests passed for all of the original implementations, so I judge them to be equivalent to the specifications. The original issue #21810 recommended reimplementing all of these functions, but I think that recommendation was based on the premise that the functions *might* be incorrect. 

To move things forward, how about dropping 8a96009 from this PR?